ComponentKit/Core/CKComponentController.mm

```diff
@@ -191,15 +193,31 @@ - (void)_relinquishView
 - (void)_cleanupAppliedAnimations
 {
   for (const auto &appliedAnimation : _appliedAnimationsOnInitialMount) {
-    appliedAnimation.animation.cleanup(appliedAnimation.context);
+    appliedAnimation.second.animation.cleanup(appliedAnimation.second.context);
   }
   _appliedAnimationsOnInitialMount.clear();
   for (const auto &appliedAnimation : _appliedAnimations) {
-    appliedAnimation.animation.cleanup(appliedAnimation.context);
+    appliedAnimation.second.animation.cleanup(appliedAnimation.second.context);
   }
   _appliedAnimations.clear();
 }
 
+- (void)_applyAnimation:(const CKPendingComponentAnimation &)pendingAnimation toMap:(CKAppliedComponentAnimationMap &)map
+{
+  [CATransaction begin];
+  CKComponentAnimationID animationID = _nextAnimationID++;
+  [CATransaction setCompletionBlock:^() {
+    auto it = map.find(animationID);
```

Comment:
Good catch! I hadn't noticed that. We can make the block capture the controller, just to make sure this will work. It's a bit obscure, though, and I'd rather have the code duplicated.
Other possibility is to turn CKAppliedComponentAnimationMap into a std::shared_ptr and pass it by copy. Not sure I like that either.